src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -1116,6 +1116,14 @@ public SolutionState AddAnalyzerReference(ProjectId projectId, AnalyzerReference
                 throw new ArgumentNullException(nameof(analyzerReference));
             }
 
+            //PROTOTYPE
+            if (_projectIdToGeneratorDriverMap.ContainsKey(projectId))
+            {
+                var driver = _projectIdToGeneratorDriverMap[projectId];
```

Comment:
Please consider using `TryGetValue()` instead to avoid the extract look up. Same comment for methods below.